src/components/PeoplePage/PeoplePage.tsx

```diff
@@ -0,0 +1,100 @@
+import { useEffect, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { Loader } from '../Loader';
+import { Person } from '../../types';
+import { getPeople } from '../../api';
+import { PeopleTable } from '../PeopleTable';
+import { ErrorTypes } from '../../enum/ErrorTypes';
+import { PeopleFilters } from '../PeopleFilters/PeopleFilters';
+import { getFilteredPeople } from '../../utils/getFilteredPeople';
+import { getSortedPeople } from '../../utils/getSortedPeople';
+
+export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<ErrorTypes>(ErrorTypes.None);
+  const [showTable, setShowTable] = useState(false);
+  const [searchParams] = useSearchParams();
+  const sort = searchParams.get('sort');
+  const order = searchParams.get('order');
+
+  const loadPeople = async () => {
+    try {
+      setLoading(true);
+      setError(ErrorTypes.None);
+      const peopleFromServer = await getPeople();
+
+      setPeople(peopleFromServer.map(person => ({
+        ...person,
+        mother: peopleFromServer.find(
+          mother => mother.name === person.motherName,
+        ),
+        father: peopleFromServer.find(
+          father => father.name === person.fatherName,
+        ),
+      })));
+
+      if (!peopleFromServer) {
```

Comment:
This condition check should be placed before the 'setPeople' function is called, to prevent unnecessary operations when peopleFromServer is null or undefined.